packages/menu/src/SubMenu.tsx

```diff
@@ -0,0 +1,349 @@
+    color: ${uiColors.blue.dark3};
+  }
+`;
+
+const activeMainTextStyle = css`
+  font-weight: bolder;
+  color: ${uiColors.green.dark3};
+`;
+
+const iconButtonStyle = css`
+  position: absolute;
+  z-index: 1;
+  right: 8px;
+  top: ${52 / 2 - 22 / 2}px;
+  margin: auto;
+  background-color: ${uiColors.gray.light3};
+  transition: background-color 150ms ease-in-out;
+  ${subMenuContainer.selector}:hover + & {
+    background-color: ${uiColors.gray.light2};
+    transition: background-color 150ms ease-in-out;
+  }
+`;
+
+const iconButtonFocusedStyle = css`
+  ${subMenuContainer.selector}:focus + & {
+    background-color: ${uiColors.blue.light3};
+    transition: background-color 150ms ease-in-out;
+  }
+`;
+
+const openIconButtonStyle = css`
+  background-color: ${uiColors.white};
+  transition: background-color 150ms ease-in-out;
+`;
+
+const mainIconStyle = css`
+  color: ${uiColors.gray.base};
+  // 15px of padding from MenuItem + 32px width of SVG + 13px = 60px
+  // which is the padding-left for children of SubMenus.
+  margin-right: 13px;
+  flex-shrink: 0;
+  ${subMenuContainer.selector}:hover > & {
+    color: ${uiColors.gray.dark1};
+  }
+`;
+
+const mainIconFocusedStyle = css`
+  ${subMenuContainer.selector}:focus > & {
+    color: ${uiColors.blue.base};
+  }
+`;
+
+const activeIconStyle = css`
+  color: ${uiColors.green.base};
+  ${subMenuContainer.selector}:hover > & {
+    color: ${uiColors.green.base};
+  }
+`;
+
+const ulStyle = css`
+  list-style: none;
+  padding: 0;
+  height: 0;
+  overflow: hidden;
+  transition: height 150ms ease-in-out;
+`;
+
+const menuItemBorder = css`
+  position: absolute;
+  width: 100%;
+  height: 1px;
+  background: ${uiColors.gray.light2};
+  top: 0;
+`;
+
+type Glyph = keyof typeof glyphs;
+
+interface SharedSubMenuProps {
+  open?: boolean;
+  setOpen?: React.Dispatch<React.SetStateAction<boolean>>;
+  className?: string;
+  description: React.ReactElement;
+  disabled?: boolean;
+  active?: boolean;
+  glyph?: Glyph;
+  onExited?: ExitHandler;
+}
+
+interface LinkSubMenuProps extends HTMLElementProps<'a'>, SharedSubMenuProps {
+  href: string;
+}
+
+interface ButtonSubMenuProps
+  extends HTMLElementProps<'button'>,
+    SharedSubMenuProps {
+  href?: null;
+}
+
+type SubMenuProps = LinkSubMenuProps | ButtonSubMenuProps;
+
+function usesLinkElement(
+  props: LinkSubMenuProps | ButtonSubMenuProps,
+): props is LinkSubMenuProps {
+  return props.href != null;
+}
+
+const subMenuItemHeight = 36;
+
+const SubMenu = React.forwardRef((props: SubMenuProps, ref) => {
+  const {
+    title,
+    description,
+    href,
+    children,
+    setOpen,
+    onKeyDown,
+    className,
+    onClick,
+    glyph,
+    onExited = () => {},
+    open = false,
+    active = false,
+    disabled = false,
+    ...rest
+  } = props;
+
+  const { usingKeyboard: showFocus } = useUsingKeyboardContext();
+
+  const iconButtonRef: React.MutableRefObject<HTMLElement | null> = useRef(
```

Comment:
I don't think I've seen `MutableRefObject` used. Why that vs `refObject`?

Reply:
👍